velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -25,6 +25,17 @@
 
 namespace facebook::velox::functions::sparksql {
 
+std::shared_ptr<DateTimeFormatter> getDateTimeFormatter(
+    bool legacyTimeParser,
```

Comment:
We are building a formatter, but take a parameter called 'parser'. Aren't these the opposite actions? 

We also take 'lenient' boolean, which seems to apply only to legacyTimeParser.

It might be clearer to change the order of inputs to: format, legacy, lenient.

Also, we use term 'simple' for 'legacy' elsewhere. Perhaps, rename legacy to simple. Or, better yet, replace 2 booleans with DateTimeFormatterType.

Reply:
Thanks! Updated. This is much better.